changelog/@unreleased/pr-258.v2.yml

```diff
@@ -0,0 +1,5 @@
+type: improvement
+improvement:
+  description: Only enable TCP logging if both log metadata and receiver are configured
```

Comment:
Can you expand this a bit to explain the different behaviors for the different combinations? The explanation in the PR was very helpful to understand. The changelog doesn't need to be quite as extensive, but still think it would help to have further details about the expected behavior/usage here.